### DISCUSSION TITLE
Multiupload to folders

### DISCUSSION THREAD

**Discussion by scottypres** [OctoFarm/OctoFarm#737] [Implemented]
OG patreon supporter here. It would be terrific to have multi-upload upload full directories or be able to specify a directory to upload to. Since I have 27 printers and most of them have the same directories, when I multi-upload a file, it goes to the root directory. This means I have hundreds of files in the root directory that is time consuming to go through to start prints.

If we could start a common file from multiple printers at once, this would also be great, but not as important as the first request. Thanks!

**Comment by NotExpectedYet** [comment]
Agree'd, been meaning to sort that for a while so I'll make a note to take a look at it. It can always check for the folders existence and create it if it doesn't already exist. 

The common file problem is been worked on just to let you know :). We've been thinking about a OctoFarm based file manager for that rather than scanning and combining multiple OP file lists. OctoFarm would then upload and delete the files when required. This would add a slight bit to the print time but I think the ease of management would be a good trade off.

**Reply by pat1121** [reply]
The file manager is the feature I'm most looking forward to!

**Comment by NotExpectedYet** [comment]
> OG patreon supporter here. It would be terrific to have multi-upload upload full directories or be able to specify a directory to upload to. Since I have 27 printers and most of them have the same directories, when I multi-upload a file, it goes to the root directory. This means I have hundreds of files in the root directory that is time consuming to go through to start prints.
> 
> If we could start a common file from multiple printers at once, this would also be great, but not as important as the first request. Thanks!

Starting a common file will be a feature of the group view currently. Once I've done that implementation I'll have a think about porting it over to the other views. 

I've however, re-hauled the multi-upload function to include folder creation/checking now, there is also a new Multi-Print option on the views. These will be available in V1.2.
